src/goose/brokers/redis/batch.clj

```diff
@@ -0,0 +1,27 @@
+(ns goose.brokers.redis.batch
+  (:require
+    [goose.brokers.redis.commands :as redis-cmds]
+    [goose.brokers.redis.retry :as retry]
+    [goose.defaults :as d]))
+
+(defn metadata-update [next]
+  (fn [{:keys [redis-conn]
+        :as   opts}
+       {:keys [id, batch-id]} job]
+    (if (contains? job :batch-id)
+      (try
+        (let [response (next opts job)
+              _ (redis-cmds/del-and-add redis-conn
+                                        (d/batch-enqueued-set batch-id)
+                                        (d/batch-successful-set batch-id)
+                                        id)]
+          response)
+        (catch Exception ex
+          (let [failed-job (goose.retry/set-failed-config job ex)]
+            (if (retry/max-retries-reached? failed-job)
+              (redis-cmds/del-and-add redis-conn
+                                      (d/batch-enqueued-set batch-id)
+                                      (d/batch-failed-set batch-id)
```

Comment:
s/failed/dead? We're using dead-jobs terminology across Goose for jobs exhausting retries.

Reply:
Are you suggesting to put the function in spec?

Reply:
I am suggesting renaming batch-failed-set to batch-dead-set, because jobs who have exhausted retries are labelled as dead inside goose.